src/test/fio/fio_ceph_messenger.cc

```diff
@@ -300,8 +301,9 @@ static Messenger *create_messenger(struct ceph_msgr_options *o)
     ceph_msgr_types[o->ms_type] :
     g_ceph_context->_conf.get_val<std::string>("ms_type");
 
+  auto nonce = o->is_receiver ? 0 : ceph::util::generate_random_number<uint64_t>();
   Messenger *msgr = Messenger::create(g_ceph_context, ms_type.c_str(),
-				      ename, lname, 0, flags);
+				      ename, lname, nonce, flags);
```

Comment:
how about using the pid?

Reply:
i firstly used getpid(). But for multi threads in a process, getpid() can't work. So i change to use random.

Reply:
i see. can we use `o.td__.thread_number` as a unique number (nonce) representing the client? and we can even mix it with `o.td__.pid` for the nonce if you think we should be able to start multiple fio client on a single host.

Reply:
see https://github.com/axboe/fio/blob/971341ad741db8d2ae57d8f9b368d75f6c02c1a9/fio.h#L183

Reply:
@tchaikov . i print o->td__->pid, this value is zero.  BTW why you did not like nonce by random?

Reply:
@majianpeng what about the `thread_number`?

> BTW why you did not like nonce by random?

the random nonce just works from practical perspective. but i am after a better alternative to ensure it's unique.

Reply:
@tchaikov . we can use thread_numb + getpid(). at this point, thread_number already set.